### PR TITLE
Moving `hasExportsOfType` to image service.

### PR DESCRIPTION
**Pull request by akash1810**
sorry for the big diff...

### CODE REVIEW

File: kahuna/public/js/image/service.js [hunk 1]
```diff
@@ -28,6 +29,18 @@ imageService.factory('imageService', ['editsService', function(editsService) {
         return { data, save, remove };
     }
 
+    function hasExportsOfType(image, type) {
+        return image.data.exports &&
+                image.data.exports.some(ex => ex.type === type);
+    }
+
+    function states(image) {
```
Comment (theefer) [inline kahuna/public/js/image/service.js]: Could we stick to making functions verbs describing their action?

File: kahuna/public/js/preview/image.html [hunk 2]
```diff
@@ -66,8 +66,8 @@
 
             <ui-archiver
                 class="bottom-bar__action"
-                image="image"
-                disabled="selectionMode"></ui-archiver>
+                image="ctrl.image"
+                disabled="ctrl.selectionMode"></ui-archiver>
 
             <div class="preview__cost bottom-bar__action" ng:switch="image.data.cost">
```
Comment (theefer) [inline kahuna/public/js/preview/image.html]: I think you missed one `image` here and below.

Comment (akash1810) [inline kahuna/public/js/preview/image.html]: oops, fixed now.

File: kahuna/public/js/image/service.js [hunk 3]
```diff
@@ -7,7 +7,8 @@ export const imageService = angular.module('gr.image.service', ['kahuna.edits.se
 imageService.factory('imageService', ['editsService', function(editsService) {
     function forImage(image) {
         return {
-            usageRights: usageRights(image)
+            usageRights: usageRights(image),
+            states: getStates(image)
```
Comment (jamesgorrie) [inline kahuna/public/js/image/service.js]: Not sure states sounds right here i.e. It's in the hasCrops state?
I can't think of anything better though :bike:...

**Comment by jamesgorrie** [comment]
:+1:

**Pull request merged**